com.unity.render-pipelines.universal/ShaderLibrary/ShaderDebugPrint.hlsl

```diff
@@ -0,0 +1,279 @@
+#ifndef SHADER_DEBUG_PRINT_INCLUDED
+#define SHADER_DEBUG_PRINT_INCLUDED
+
+// Include this header to any shader to enable debug printing values from shader code to console
+//
+// Example:
+// float4 colorRGBA = float4(0.1, 0.2, 0.3, 0.4);
+// if(all(int2(pixel.xy) == int2(100, 100))
+//     ShaderDebugPrint(ShaderDebugTag('C','o','l'), colorRGBA);
+// ----
+// Output:
+// Frame #270497: Col  float4(0.1f, 0.2f, 0.3f, 0.4f)
+
+// Output buffer bound into "last" slot by convention
+RWStructuredBuffer<uint> shaderDebugOutputData : register(u7);
+
+
+static const uint MaxShaderDebugOutputElements = 1024 * 16; // 16KB - must match the C# side buffer size (16KB / 4 (bytes per uint) / 6 (uints, header+tag+payload) == 682 uint4s)
+// Input Constants
+float4 _ShaderDebugPrintInputMouse;
```

Comment:
This should probably be wrapped in a CBUFFER() macro - otherwise the shader will not be SRP batcher compatible.

Reply:
Should be fixed now.